calculator_test.go

```diff
@@ -2,23 +2,407 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
+	"reflect"
+	"runtime"
 	"testing"
+	"time"
 )
 
-func TestAdd(t *testing.T) {
+// Note Fatal|FatalF are also useful methods,
+// to stop further execution within the same test.
+
+// Return the name of a function.
+func getFuncName(i interface{}) string {
+	return runtime.FuncForPC(reflect.ValueOf(i).Pointer()).Name()
+}
+
+// Return a slice of N random numbers from 0-500
+func randFloat64Slice(n int) []float64 {
+	rand.Seed(time.Now().UnixNano())
+	r := make([]float64, n)
+
+	// This `range` serves to loop the right number of times for the size of r
+	for i := range r {
+		x := rand.Float64() * float64(rand.Intn(500))
+		r[i] = x
+	}
+	return r
+}
+
+// Test Add(), Subtract(), and Multiply()
+func TestAddSubtractMultiply(t *testing.T) {
+	// Define test cases
+	// This anonymous struct has no name, as its only used here.
+	testCases := []struct {
+		function    func(float64, float64, ...float64) float64
+		a           float64
+		b           float64
+		want        float64
+		description string
+	}{
+		{
+			function:    calculator.Add,
+			description: "two positive numbers which sum to a positive",
+			a:           2,
+			b:           2,
+			want:        4,
+		},
+		{
+			function:    calculator.Add,
+			description: "a positive and negative number which sum to a positive",
+			a:           7,
+			b:           -2,
+			want:        5,
+		},
+		{
+			function:    calculator.Add,
+			description: "a positive and negative number which sum to a negative",
+			a:           3,
+			b:           -5,
+			want:        -2,
+		},
+		{
+			function:    calculator.Subtract,
+			description: "two positive numbers whos difference is negative",
+			a:           2,
+			b:           9,
+			want:        -7,
+		},
+		{
+			function:    calculator.Subtract,
+			description: "two positive numbers whos difference is positive",
+			a:           7,
+			b:           2,
+			want:        5,
+		},
+		{
+			function:    calculator.Subtract,
+			description: "one positive and one negative decimal number whos difference is a positive decimal",
+			a:           3,
+			b:           -2.5,
+			want:        5.5,
+		},
+		{
+			function:    calculator.Multiply,
+			description: "two positive numbers whos product is positive",
+			a:           2,
+			b:           20,
+			want:        40,
+		},
+		{
+			function:    calculator.Multiply,
+			description: " a positive and negative number whos product is negative",
+			a:           7,
+			b:           -2,
+			want:        -14,
+		},
+		{
+			function:    calculator.Multiply,
+			description: "a positive decimal and negative decimal whos product is a negative decimal",
+			a:           8.4,
+			b:           -2.5,
+			want:        -21,
+		},
+	}
+
+	t.Parallel()
+
+	for _, c := range testCases {
+		got := c.function(c.a, c.b)
+		if got != c.want {
```

Comment:
It's a good idea to always compare `want` and `got` in the same order, and since it makes sense to us to output the _difference_ as 'want X, got Y', we should do the comparison in the same way for symmetry, shouldn't we?

```suggestion
		if c.want != got {
```